drivers/misc/mcux_flexio/mcux_flexio.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2024, STRIM, ALC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_MCUX_FLEXIO_H_
+#define ZEPHYR_DRIVERS_MCUX_FLEXIO_H_
+
+#include <zephyr/device.h>
+
+/**
+ * @struct mcux_flexio_child_res
+ * @brief Structure containing information about the required
+ * resources for a FlexIO child.
+ */
+struct mcux_flexio_child_res {
+	uint8_t *shifter_index;
+	uint8_t shifter_count;
+	uint8_t *timer_index;
+	uint8_t timer_count;
+};
```

Comment:
Is this driver not managing the pins anymore?

Reply:
The previous version of your implementation had an entry for the pins. I guess the expectation is that FlexIO pins are managed by the individual drivers based on the board dts file. Am I understanding this correctly?

Reply:
@mmahadevan108, Yes, the selection and configuration of pins is determined in dts using `sdo-pin`, `sdi-pin`, `sck-pin`, `cs-gpios` and `pinctrl`. Any shifter or timer can be mixed with any pin, so for convenience, their assignment is handled by the parent driver.